mixin.scss

```diff
@@ -0,0 +1,32 @@
+@function bk-calculateTypeOffset($line-height, $font-size, $descdender-height-scale){
+  $line-height-scale: $line-height / $font-size;
+  $line-height-scale: (($line-height-scale - 1) / 2) + $descdender-height-scale;
+  @return $line-height-scale;
+}
+
+@mixin bk-render($font-size, $line-height, $type-offset){
+  font-size: $font-size * 1px;
+  line-height: $line-height * 1px;
+  transform: translateY( $type-offset * 1em );
+}
+
+@mixin basekick(
+  $bk-type-size-modifier,
+  $bk-type-row-span,
+  $bk-descender-height-scale,
+  $bk-base-font-size,
+  $bk-grid-row-height,
+  $bk-line-height-override: false){
+
+  $font-size: $bk-type-size-modifier * $bk-base-font-size;
+
+  @if type-of($bk-line-height-override) == number {
+    $line-height: $bk-line-height-override;
+    $type-offset: bk-calculateTypeOffset($line-height, $font-size, $bk-descender-height-scale);
+    @include bk-render($font-size, $line-height, $type-offset);
+  } @else {
+    $line-height: $bk-grid-row-height * $bk-type-row-span;
+    $type-offset: bk-calculateTypeOffset($line-height, $font-size, $bk-descender-height-scale);
+    @include bk-render($font-size, $line-height, $type-offset);
+  }
```

Comment:
I would propose cleaning up some of this duplication by using a more ternary like syntax for the `line-height`/`line-height-override`:
```sass
$line-height: if(type-of($bk-line-height-override) == number,
  $bk-line-height-override,
  $bk-grid-row-height * $bk-type-row-span);
```
Then you would not need to duplicate the `$type-offset` calculation, and could arguably do away with the `bk-render` mixin and just have the properties inline.